src/python/upload.py

```diff
@@ -54,8 +54,20 @@ def upload(obj_dict, obj_type, excel_uploader_object, gui_object):
     for key, obj in obj_dict.items():
 
         try:
-            if obj_type == "File" and obj.name is None:
-                obj.name = obj.source
+            if obj_type == "File":
+                # iterate through list of files
+                for file in obj:
+                    if file.name is None:
+                        file.name = file.source
+                    # nested try and except to handle duplicate file issues
+                    try:
```

Comment:
well done!

I like that you are handling if it is already there, so it doesn't break the program and I love the comment too!